tests/testthat/test-tidyeval-across.R

```diff
@@ -0,0 +1,331 @@
+      lazy_frame(x = 1) %>%
+        summarise(across(x, 42))
+    ))
+    (expect_error(
+      lazy_frame(x = 1) %>%
+        summarise(across(y, mean))
+    ))
+  })
+})
+
+test_that("across() translates character vectors", {
+  lf <- lazy_frame(a = 1, b = 2)
+  expect_snapshot(lf %>% summarise(across(a:b, "log")))
+  expect_snapshot(lf %>% summarise(across(a:b, "log", base = 2)))
+
+  expect_snapshot(lf %>% summarise(across(a, c("log", "exp"))))
+
+  out <- lf %>% summarise(across(a:b, c(x = "log", y = "exp")))
+  expect_equal(colnames(out), c("a_x", "a_y", "b_x", "b_y"))
+})
+
+test_that("across() translates functions", {
+  lf <- lazy_frame(a = 1, b = 2)
+  expect_snapshot(lf %>% summarise(across(a:b, log)))
+  expect_snapshot(lf %>% summarise(across(a:b, log, base = 2)))
+
+  expect_snapshot(lf %>% summarise(across(a:b, list(log, exp))))
+
+  out <- lf %>% summarise(across(a:b, list(x = log, y = exp)))
+  expect_equal(colnames(out), c("a_x", "a_y", "b_x", "b_y"))
+})
+
+test_that("untranslatable functions are preserved", {
+  lf <- lazy_frame(a = 1, b = 2)
+  expect_snapshot(lf %>% summarise(across(a:b, SQL_LOG)))
+})
+
+test_that("across() translates formulas", {
+  lf <- lazy_frame(a = 1, b = 2)
+  expect_snapshot(lf %>% summarise(across(a:b, ~ log(.x, 2))))
+  expect_snapshot(lf %>% summarise(across(a:b, list(~ log(.x, 2)))))
+})
+
+test_that("across() translates NULL", {
+  lf <- lazy_frame(a = 1, b = 2)
+  expect_snapshot(lf %>% mutate(across(a:b)))
+})
+
+test_that("across() can rename", {
+  expect_equal(
+    lazy_frame(x = 1:10) %>%
+      mutate(across(.cols = c(y = x), .fns = as.character)) %>%
+      op_vars(),
+    c("x", "y")
+  )
+})
+
+test_that("across() works with .cols = character()", {
+  lf <- lazy_frame(x = 1:10)
+
+  expect_equal(
+    lf %>% mutate(across(.cols = character(), .fns = as.character)),
+    lf
+  )
+})
+
+test_that("partial_eval_dots() names automatically", {
+  expect_named(
+    partial_eval_dots(lazy_frame(a = 1, b = 2), across(everything())),
+    c("a", "b")
+  )
+})
+
+test_that("across() correctly names output columns", {
+  gf <- lazy_frame(x = 1, y = 2, z = 3, s = "") %>% group_by(x)
+
+  expect_equal(
+    summarise(gf, across()) %>% op_vars(),
+    c("x", "y", "z", "s")
+  )
+  expect_equal(
+    summarise(gf, across(.names = "id_{.col}")) %>% op_vars(),
+    c("x", "id_y", "id_z", "id_s")
+  )
+  expect_equal(
+    summarise(gf, across(1:2, mean)) %>% op_vars(),
+    c("x", "y", "z")
+  )
+  expect_equal(
+    summarise(gf, across(1:2, mean, .names = "mean_{.col}")) %>% op_vars(),
+    c("x", "mean_y", "mean_z")
+  )
+  expect_equal(
+    summarise(gf, across(1:2, list(mean = mean, sum = sum))) %>% op_vars(),
+    c("x", "y_mean", "y_sum", "z_mean", "z_sum")
+  )
+  expect_equal(
+    summarise(gf, across(1:2, list(mean = mean, sum = sum), .names = "{.fn}_{.col}")) %>% op_vars(),
+    c("x", "mean_y", "sum_y", "mean_z", "sum_z")
+  )
+
+  skip("not yet decided")
+  expect_equal(
```

Comment:
Here, `dbplyr` is smarter than `dplyr` and automatically creates names. What do you prefer?

Reply:
Probably best to be consistent with dplyr.